src/precompile.jl

```diff
@@ -234,28 +234,32 @@ function _precompile_manual_()
     #   @assert Base.precompile(Tuple{Core.kwftype(typeof(Trixi.Type)),NamedTuple{(:interval, :extra_analysis_integrals),Tuple{Int,Tuple{typeof(entropy)}}},Type{AnalysisCallback},equations_type,DG{RealT,LobattoLegendreBasis{RealT,nnodes_,Array{RealT,2},StaticArrays.SArray{Tuple{4,2},RealT,2,2*nnodes_},StaticArrays.SArray{Tuple{nnodes_,nnodes_},RealT,2,nnodes_^2}},Trixi.LobattoLegendreMortarL2{RealT,nnodes_,StaticArrays.SArray{Tuple{nnodes_,nnodes_},RealT,2,nnodes_^2}},Function,Trixi.AbstractVolumeIntegral}})
     # end
   end
-  Base.precompile(Tuple{typeof(SummaryCallback)})
+  @assert Base.precompile(Tuple{typeof(SummaryCallback)})
```

Comment:
To make sure that we don't include failing precompile statements, cf. https://timholy.github.io/SnoopCompile.jl/stable/snoopi/.

Reply:
OK. In that case I think we should add a comment to that file (and maybe to the place where we include the precompile statements) that says that if this fails for any reason, it is safe to just disable the failing call to `precompile` (or to not include `precompile.jl` at all). Otherwise this is very hard to debug for new users who do something that "destroys" the precompilation.

Reply:
Done in 5486c22.